samtranslator/model/api/api_generator.py

```diff
@@ -269,6 +288,69 @@ def _add_auth(self):
         # Assign the Swagger back to template
         self.definition_body = swagger_editor.swagger
 
+    def _add_gateway_responses(self):
+        """
+        Add Gateway Response configuration to the Swagger file, if necessary
+        """
+
+        if not self.gateway_responses:
+            return
+
+        if self.gateway_responses and not self.definition_body:
+            raise InvalidResourceException(
+                self.logical_id, "GatewayResponses works only with inline Swagger specified in "
+                                 "'DefinitionBody' property")
+
+        # Make sure keys in the dict are recognized
+        for responses_key, responses_value in self.gateway_responses.items():
+            if responses_key not in GatewayResponsesProperties._fields:
+                raise InvalidResourceException(
+                    self.logical_id, "Invalid key '{}' for 'GatewayResponses' property".format(responses_key))
+
+            if 'StatusCode' not in responses_value.keys():
```

Comment:
If I'm not looking at this wrong, StatusCode should be optional.
It should only be provided if you want to change the original status code:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-gatewayresponse.html#cfn-apigateway-gatewayresponse-statuscode

Reply:
You are right. I was developing against [this documentation][], which doesn’t mention that. When I get back to my desk, I'll compare what I wrote to the CF docs to see if there's anything else like this. 

[this documentation]: 
https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-gateway-responses.gatewayResponse.html

Reply:
Done!